### PR TITLE
Address GitHub Super Linter, detekt, and unit test issues

### PR DESCRIPTION
**Pull request by jaredstockton**
Getting to the final stages of setting up a test GitHub actions pipeline (on another repo mirroring this one) and it detected some failures. I've remediated them here to allow the validation scans to succeed once merged :)

### CODE REVIEW

**Comment by jaredstockton** [comment]
@BottleRocket-Colin As mentioned in a comment on your PR, I'll wait for yours to merge prior and make adjustments to this PR once I bring in the latest changes from develop at that point in time

**Comment by jaredstockton** [comment]
@BottleRocket-Colin Merged in the latest changes from develop. Ready for review!

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/snippet/SnippetDetailsViewModel.kt [hunk 1]
```diff
@@ -156,31 +156,33 @@ class SnippetDetailsViewModel : BaseViewModel() {
...
-    //                     parent.childrenComments.add(child)
-    //                     toBeRemoved.add(child)
-    //                 }
-    //             }
-    //             unsortedComments.removeAll(toBeRemoved)
-    //             recursiveSort(parent.childrenComments, unsortedComments)
-    //         }
-    //     }
-    // }
+    @Suppress("UnusedPrivateMember")
```
Comment (BottleRocket-Colin) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/snippet/SnippetDetailsViewModel.kt:159]: Better solution than my comments, like it.

File: compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawerItem.kt [hunk 2]
```diff
@@ -76,7 +76,7 @@ fun NavItemPreview() {
 
 @Preview(showBackground = true)
 @Composable
-fun NavItemSelectedPreview() {
+private fun NavItemSelectedPreview() {
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawerItem.kt:79]: This indeed is a "Super" linter :)

**Review by BottleRocket-Colin — APPROVED** [review]

**Pull request merged**